### ISSUE TITLE
MenuItem as Link not currently working

### ISSUE THREAD

**Issue by cstrat** [mantinedev/mantine#7]
Unless undocumented and different to Button - I can't seem to get a `MenuItem` to behave like a React Router Link element. This works fine with `Button` elements. As per the docs:

https://mantine.dev/core/button/#usage-with-react-router-and-other-libraries

````
<MenuItem component={Link} to="/account" icon={<FontAwesomeIcon fixedWidth icon={faUserCog} />}>Account Settings</MenuItem>
````

This is key as my project uses `MenuItem` as React Router links in my application.

I looked at the source but have zero typescript experience so am a little confused. It almost looks like `MenuItem` is based on `Button` but it doesn't accept the props for `component`, `to` etc...

**Comment by cstrat** [comment]
Looking further into the docs, this should work!

All other props are supposed to be passed through to the `MenuItem` which is just a `Button` under the hood.

![image](https://user-images.githubusercontent.com/4209076/116966660-0624ba00-acf4-11eb-8355-5aa0c4e1a095.png)

**Comment by rtivital** [comment]
Hi, I guess, it was not clear from documentation – MenuItem renders regular button element, not Button component, I've updated the docs to clarify that part.

MenuItem must always render `button` element to support keyboard events (handle Enter and Space key presses), if you need to change a page with menu, use `() => { window.location.href = '/target-url'; }` for regular links or `() => history.push('/target-url')` for react-router with `onClick` handler.

**Comment by cstrat** [comment]
Thanks for clarifying & providing a solution.
Yeah I wasn't sure - and it does get confusing with the `button` vs `Button`

I am refactoring parts of my application from evergreen-ui which has some similar concepts. This works with their menu items while still supporting keyboard events.

**Comment by cstrat** [comment]
So this is actually proving to be a little bit annoying.
It breaks the user being able to middle click to launch a new tab etc...
I don't fully understand why you can support this:

````
<Button component={Link} to="/react-router">
    React router link
</Button>
````

But not on `MenuItem`?

Don't you need to support the same keyboard events with a `<Button>` as you do with `button`?

**Comment by rtivital** [comment]
Well, okay, let's add it, seems reasonable

**Comment by rtivital** [comment]
Resolved in [1.1.0](https://github.com/mantinedev/mantine/releases/tag/1.1.0)

**Comment by lucyxyl** [comment]
I'm struggling to get this to work. @rtivital , how should the code look? Like this?

```
<Link href={`/dash/customers/${item.id}`}>
    <MenuItem>View Details</MenuItem>
</Link>
```

I've tried this and other variants, and nothing works.
